src/components/features/gift-order/ThemeProductSection.tsx

```diff
@@ -0,0 +1,172 @@
+import { useState, useEffect } from 'react';
+import { useFetch } from '@/hooks/useFetch';
+import { getAuthTokenFromSession, BASE_URL } from '@/constants/api';
+import { Spinner } from '@/components/shared/ui/Spinner';
+import styled from '@emotion/styled';
+import { theme } from '@/styles/theme';
+import type { Product } from '@/types';
+import { useInfiniteScroll } from '@/hooks/useInfiniteScroll';
+
+interface ThemeProductSectionProps {
+  themeId: number;
+  onProductClick: (product: Product) => void;
+}
+
+const LIMIT = 20;
+
+type ProductListResponse = {
+  list: Product[];
+  hasMoreList: boolean;
+};
+
+export function ThemeProductSection({
+  themeId,
+  onProductClick,
+}: ThemeProductSectionProps) {
+  const token = getAuthTokenFromSession();
+  const [cursor, setCursor] = useState(0);
+  const [products, setProducts] = useState<Product[]>([]);
+  const [hasMore, setHasMore] = useState(true);
+
+  const { data, loading, error } = useFetch<ProductListResponse>({
+    baseUrl: BASE_URL,
+    path: `/api/themes/${themeId}/products`,
+    searchParams: { cursor: String(cursor), limit: String(LIMIT) },
+    headers: { Authorization: token },
+    deps: [themeId, cursor],
+    skip: !hasMore,
+  });
+
+  useEffect(() => {
+    if (!data) return;
+    if (cursor === 0) {
+      setProducts(data.list || []);
+    } else {
+      setProducts(prev => [...prev, ...(data.list || [])]);
+    }
+    setHasMore(data.hasMoreList);
+  }, [data, cursor]);
+
+  const handleIntersect = () => {
+    if (hasMore && !loading) {
+      setCursor(prev => prev + LIMIT);
+    }
+  };
+  const loader = useInfiniteScroll(handleIntersect, hasMore && !loading);
+
+  if (loading && products.length === 0) return <Spinner />;
+  if (error) return <ProductContainer>에러가 발생했습니다.</ProductContainer>;
+  if (!loading && products.length === 0)
+    return <ProductContainer>상품이 없습니다.</ProductContainer>;
+
+  return (
+    <ProductContainer>
+      <ProductGrid>
+        {products.map(product => (
+          <ProductCard key={product.id} onClick={() => onProductClick(product)}>
+            <ProductImage src={product.imageURL} alt={product.name} />
+            <ProductInfo>
+              <BrandName>{product.brandInfo.name}</BrandName>
+              <ProductName>{product.name}</ProductName>
+              <ProductPrice>
+                {product.price.sellingPrice.toLocaleString()}원
+              </ProductPrice>
+            </ProductInfo>
+          </ProductCard>
+        ))}
+      </ProductGrid>
+      <div ref={loader} style={{ height: 20 }} />
```

Comment:
[제안] 아래처럼 내부에서 useInfiniteScroll를 사용하는 컴포넌트화해도 좋을 것 같아요
```suggestion
      <Loader onView={handleIntersect} style={{ height: 20 }} />
```